<?xml version="1.0" encoding="UTF-8"?>
<!-- Launch the joy node and its translator.
     Written by Tiger Sachse and Harrison Black. -->
<launch>
  <arg name="wheel_instructions_topic" doc="The topic for wheel instructions."/>
  <arg name="front_arm_instructions_topic" doc="The topic for front arm instructions."/>
  <arg name="back_arm_instructions_topic" doc="The topic for back arm instructions."/>
  <arg name="front_drum_instructions_topic" doc="The topic for front drum instructions."/>
  <arg name="back_drum_instructions_topic" doc="The topic for back drum instructions."/>
  <arg name="max_wheel_speed" doc="The robot's wheels' max speed."/>
  <arg name="max_arm_speed" doc="The robot's arms' max speed."/>
  <arg name="max_drum_speed" doc="The robot's drums' max speed."/>
  <arg name="joystick_device"
       default="default"
       doc="The joystick device that Joy should listen to."/>

  <group if="$(eval joystick_device == 'default')">
    <rosparam command="load" file="$(find ezrassor_joy_translator)/config/joy_translator.yaml"/>
    <node name="joy_node" pkg="joy" type="joy_node" output="screen"/>
  </group>
  <group if="$(eval joystick_device != 'default')">
    <node name="joy_node" pkg="joy" type="joy_node" output="screen">
      <param name="dev" value="$(arg joystick_device)"/>
    </node>
  </group>

  <node name="joy_translator"
        pkg="ezrassor_joy_translator"
        type="joy_translator"
        output="screen">
    <param name="wheel_instructions_topic" value="$(arg wheel_instructions_topic)"/>
    <param name="front_arm_instructions_topic" value="$(arg front_arm_instructions_topic)"/>
    <param name="back_arm_instructions_topic" value="$(arg back_arm_instructions_topic)"/>
    <param name="front_drum_instructions_topic" value="$(arg front_drum_instructions_topic)"/>
    <param name="back_drum_instructions_topic" value="$(arg back_drum_instructions_topic)"/>
    <param name="max_wheel_speed" value="$(arg max_wheel_speed)"/>
    <param name="max_arm_speed" value="$(arg max_arm_speed)"/>
    <param name="max_drum_speed" value="$(arg max_drum_speed)"/>
  </node>
</launch>
